### PR TITLE
Migrate docs platform to Docusaurus

### PR DESCRIPTION
**Pull request by pabloest**
This PR migrates the docs from [Mkdocs](https://www.mkdocs.org/) to [Docusaurus](https://docusaurus.io/).

## Motivations

The motivations for this change include:

- enabling a sidebar navigation with our desired nav hierarchy
- basing the docs on a React-based platform for better integration with the rest of semgrep.dev
- improving the look and feel, particularly for mobile devices
- enabling unified search between the docs and the Semgrep Registry in the future

### Deployment

As with Mkdocs, deployment is handled by a CircleCI workflow [`config.yml`](https://github.com/returntocorp/semgrep-docs/compare/pablo/docusaurus-migration?expand=1#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47) file. I wasn’t able to test it locally, so I’d appreciate some extra eyes on this.

### Search

Search is based on Algolia. Algolia’s index should be updated any time there’s a content change, and this is performed by the [`algolia-scape.yml`](https://github.com/returntocorp/semgrep-docs/blob/e81ef3d3f5b242e6eb69014e4603b41227e157c0/.github/workflows/algolica-scrape.yml) GitHub Action. As this Action runs on pushes to main, the index should get updated automatically after a docs PR is merged to main. I’ve only tested this locally.

### Desktop 💻

Screen Shot 2021-07-15 at 3 43 47 PM

Screen Shot 2021-07-15 at 3 43 54 PM

### Mobile 🤳

Screen Shot 2021-07-15 at 3 44 06 PM

Screen Shot 2021-07-15 at 3 44 15 PM

Screen Shot 2021-07-15 at 3 44 19 PM

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,44 +3,53 @@ version: 2.1
...
       - run:
-          name: install aws cli
-          command: |
-            curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
-            unzip awscliv2.zip
-            sudo ./aws/install
-      - run: aws --version
-      - run: aws s3 sync site/ s3://staging-semgrep-dev/docs
+          name: Install Deps
+          command: yarn install
```
Comment (brendongo) [inline .circleci/config.yml:33]: Do you mean `yarn build`?

Comment (brendongo) [reply .circleci/config.yml:33]:
```suggestion
          command: yarn build
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,44 +3,53 @@ version: 2.1
...
-      - run: pip install mkdocs==1.2.1 mkdocs-redirects mkdocs-meta-descriptions-plugin
-      - run: mkdocs build
+      - restore_cache:
+          key: dependency-cache-{{ checksum "yarn.lock" }}
+      - run:
+          name: Install aws cli
+          command: sudo apt-get -y -qq install awscli && aws --version
+      - run:
+          name: Install Deps
+          command: yarn install
```
Comment (brendongo) [inline .circleci/config.yml:49]:
```suggestion
          command: yarn build
```

**Review by brendongo — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (pabloest) [reply .circleci/config.yml:33]: Yep it was missing the build and I think it needs both.

**Review by ievans — APPROVED** [review]
Looks great. Awesome work @pabloest

**Pull request merged**